kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,143 @@
+//! Interface for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    #[allow(non_camel_case_types)]
+    pub enum ScreenColorDepth {
+        None = 0,
+        Mono = 1,
+        RGB_233 = 8,
+        RGB_565 = 16,
+        RGB_888 = 24,
+        ARGB_8888 = 32,
+    }
+}
+
+impl From<ScreenColorDepth> for usize {
+    fn from(color_depth: ScreenColorDepth) -> usize {
+        match color_depth {
+            ScreenColorDepth::None => 0,
+            ScreenColorDepth::Mono => 1,
+            ScreenColorDepth::RGB_233 => 8,
+            ScreenColorDepth::RGB_565 => 16,
+            ScreenColorDepth::RGB_888 => 16,
+            ScreenColorDepth::ARGB_8888 => 32,
+        }
+    }
+}
+
+pub trait Screen {
+    /// Sets the screen resolution (in pixels). Returns ENOSUPPORT if the resolution is
+    /// not supported. The function should return SUCCESS for at least one resolution.
```

Comment:
Is this function synchronous? The later ones specify explicitly that they are.

If it's async, this call should probably have no return value (as the `ENOSUPPORT` will come from the complete call)

Reply:
I detailed the functionality in the documentation.